### PR TITLE
fix for TPC-TOF matching

### PR DESCRIPTION
**Pull request by noferini**
Hi Ruben,
this is the fix I prepared and here the dpl I used to test:

WORKFLOW+="o2-tof-matcher-tpc --disable-root-input -b --session default --severity info --shm-segment-id 0 --shm-segment-size 137438953472 --fairmq-rate-logging 0 --disable-mc --cosmics |"

### CODE REVIEW

**Comment by noferini** [comment]
I forgot to say that I added the subtraction of TOF latency

File: Detectors/GlobalTracking/src/MatchTOF.cxx [hunk 1]
```diff
@@ -1162,17 +1168,23 @@ void MatchTOF::doMatchingForTPC(int sec)
     minTrkTime = int(minTrkTime / BCgranularity) * BCgranularity;                                        // align min to a BC
     double maxTrkTime = (trackWork.second.getTimeStamp() + mExtraTPCFwdTime[cacheTrk[itrk]]) * 1.E6;     // maximum time in ps
 
-    /*
+    //   printf("trk time %f - %f (max shift +/- %f cm)\n",minTrkTime,maxTrkTime,trackWork.second.getTimeStampError()*vdrift );
+
+    std::array<float, 3> globalPos;
+    trackWork.first.getXYZGlo(globalPos);
+
```
Comment (shahor02) [inline Detectors/GlobalTracking/src/MatchTOF.cxx]: why is this needed? I see also in many places you do getXYZGlo (this is relatively expensive methods) just to use the result in LOG(DEBUG), i.s. almost never used. Since using LOG(DEBUG) is anyway possible only if the code was compiled with debug mode, perhaps also the getXYZGlo should be called in this case only?

**Review by shahor02 — COMMENTED** [review]
Thanks @noferini ! 

As far as I can understand, you neglect zres in chi2 calculation only because of the time bias + vdrfit uncertainty? This problems are relevant now, but in general, they should be not present in real data taking.  Perhaps one can "dilute" the zres contribution to chi2 by the extra tolerances you are to overcome these problems (and it turn, these tolerances are better to make settable).

File: Detectors/GlobalTracking/src/MatchTOF.cxx (hunk 1 above)
Comment (noferini) [reply Detectors/GlobalTracking/src/MatchTOF.cxx]: I guess it is a residual of my debug session (when I added a lot of prints and infos). Let me check

Comment (noferini) [reply Detectors/GlobalTracking/src/MatchTOF.cxx]: Removed

**Comment by noferini** [comment]
> Thanks @noferini !
> 
> As far as I can understand, you neglect zres in chi2 calculation only because of the time bias + vdrfit uncertainty? This problems are relevant now, but in general, they should be not present in real data taking. Perhaps one can "dilute" the zres contribution to chi2 by the extra tolerances you are to overcome these problems (and it turn, these tolerances are better to make settable).

This feature is enabled only for cosmics since the shift along z can be very large and vdrift may be uncalibrated. In general it is not used. I agree with all the other comments.

**Comment by shahor02** [comment]
Thanks Francesco. Do you plan to use these tolerance always in cosmics?

**Comment by noferini** [comment]
> Thanks Francesco. Do you plan to use these tolerance always in cosmics?

No, Current tolerance are quite large. I expect that more data will allow us to tune many things
